### PR TITLE
TW-1650: part II: memory footprint in web: remove unnecessary animation and Image.memory in web

### PR DESCRIPTION
**Pull request by sherlockvn**
### DESC:
#1650 

### Problem:
- Currently, we use `Image.memory` to load image data to view, which required two steps, its need to downloaded to memory first, then pass it to the widget, which make app freezes a lot while scrolling.
-  Moreover, `MxcImage` use combination of animation and setState when image is downloaded, which is not efficient.
-`Image.network` have multiple optimization, and build-in method (`loadingBuilder` and `frameBuilder`) which make the rendering in web more efficient. And We don't have to convert to bytes anymore, use the download link, and the browser will help us caching the image.

- For the case of encrypted chat, we still have to use `Image.memory` because of we have to decrypte bytes data after downloading done

- For the case of `Avatar`, use cacheWidth to reduce the memory usage when decoding image

### Reference:
[Reference of frameBuilder in Image.network](https://github.com/flutter/flutter/issues/102828)

### Compare performance and memory usage between beta.twake.app and current 
https://github.com/linagora/twake-on-matrix/assets/43041967/7469752f-37c0-4a19-a40a-38306fe92440

### Demo:

https://github.com/linagora/twake-on-matrix/assets/43041967/4723ab44-17c9-4c54-9bbf-3ee82d8d1a3a

https://github.com/linagora/twake-on-matrix/assets/43041967/841f8bf9-ff33-4ac9-be5a-c2579e99ac41

https://github.com/linagora/twake-on-matrix/assets/43041967/71f19992-0feb-4e5d-b399-f466a4aa3a41

### CODE REVIEW

File: lib/pages/chat/events/image_builder_web.dart [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            );
+          },
+        ),
+      );
+      if (result == MediaViewerPopupResultEnum.closeRightColumnFlag) {
+        closeRightColumn?.call();
+      }
+    } else if (onTapSelectMode != null) {
+      onTapSelectMode!();
+      return;
```
Comment (Te-Z) [inline lib/pages/chat/events/image_builder_web.dart:85]: since the method has a void type, no need to add return IMO

Comment (sherlockvn) [reply lib/pages/chat/events/image_builder_web.dart]: done

File: lib/pages/chat/events/image_builder_web.dart [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    } else {
+      return;
+    }
+  }
+}
+
+class UnencryptedImageWidget extends StatelessWidget {
```
[the remark this answers is not in the document]
Comment (sherlockvn) [reply lib/pages/chat/events/image_builder_web.dart]: done

File: lib/pages/chat/events/image_builder_web.dart [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return Hero(
+      tag: event.eventId,
+      child: Material(
+        child: InkWell(
+          mouseCursor: SystemMouseCursors.click,
+          borderRadius: BorderRadius.circular(12.0),
```
[the remark this answers is not in the document]
Comment (sherlockvn) [reply lib/pages/chat/events/image_builder_web.dart]: done

File: lib/pages/chat/events/images_builder/image_placeholder.dart [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        child: CircularProgressIndicator.adaptive(),
+      );
+    }
+    final String blurHashString =
+        event.blurHash ?? AppConfig.defaultImageBlurHash;
+    final ratio = event.infoMap['w'] is int && event.infoMap['h'] is int
+        ? event.infoMap['w'] / event.infoMap['h']
+        : 1.0;
+    var width = 32;
+    var height = 32;
```
Comment (nqhhdev) [inline lib/pages/chat/events/images_builder/image_placeholder.dart:34]: Move it to style file

**Review by hoangdat — APPROVED** [review]

**Comment by sherlockvn** [comment]
- Hot fix: timestamp is not updated when scroll in chat
-> move `InViewState` dispose() in chat controller, because its like a static instance, so we can't dispose it in the message dispose() method

https://github.com/linagora/twake-on-matrix/assets/43041967/59c61fd0-3c6f-47be-8fe8-809c1a472c65

File: lib/pages/chat/chat.dart [hunk 5]
```diff
@@ -1930,7 +1931,7 @@ class ChatController extends State<Chat>
     pinnedMessageScrollController.dispose();
     onUpdateEventStreamSubcription?.cancel();
     keyboardVisibilitySubscription?.cancel();
-
+    InViewNotifierListCustom.of(context)?.dispose();
```
Comment (hoangdat) [inline lib/pages/chat/chat.dart:1934]: it still dispose `inview` well?

Comment (sherlockvn) [reply lib/pages/chat/chat.dart:1934]: yes, here is where its reference.
Screenshot 2024-05-13 at 17 53 25

**Review by hoangdat — APPROVED** [review]

**Pull request merged**